include/cuda/memory_resource

```diff
@@ -158,6 +130,58 @@ _LIBCUDACXX_CONCEPT_FRAGMENT(
 template <class _Resource, class _Property, class _Return>
 _LIBCUDACXX_CONCEPT has_property_with = _LIBCUDACXX_FRAGMENT(__has_property_with_, _Resource, _Property, _Return);
 
+/// class forward_property
+/// \brief The \c forward_property crtp temaplate simplifies the user facing side of forwarding properties
+///        We can always tell people to just derive from it to properly forward all properties
+template <class _Derived, class _Upstream>
+struct forward_property {
```

Comment:
Should we add constraints that `_Derived` have a `get_upstream` function to improve diagnostics?

Otherwise you don't get a compile error until you attempt to invoke `get_property` on the Derived type: https://godbolt.org/z/f3c4EW3zs

Reply:
Now this is a tricky one.

I was pondering a constraint, but in the more common case of stateless properties, that would force the user to add a useless boilerplate function.

We can add it, but I am not too happy about it